Doppler.Sap/Controllers/BusinessPartnerController.cs

```diff
@@ -50,9 +57,9 @@ public async Task<IActionResult> CreateOrUpdateBusinessPartner([FromBody] Dopple
 
         private string VerifyUserInformation(DopplerUserDto dopplerUser)
         {
-            if (dopplerUser.BillingCountryCode != "AR")
+            if (!_sapConfig.SapServiceConfigs.Any(c => c.CountryCode == dopplerUser.BillingCountryCode))
```

Comment:
Creo que sería mejor delegar esa validación directamente al  service y en todo caso manejar correctamente la excepción, o incluso exponer un método en _BusinessPartnerService_ que haga la validación del country code.

Pero no tengo una opinión fuerte.

Reply:
Dale, ahí veo de mejorar eso y te comento lo que se me ocurra

Reply:
Te parece mover toda la logica de la validacion al servicio?. Por ejemplo:
```
private string VerifyUserInformation(DopplerUserDto dopplerUser)
        {
            if (!_sapConfig.SapServiceConfigs.Any(c => c.CountryCode == dopplerUser.BillingCountryCode))
            {
                _logger.LogInformation($"{dopplerUser.Email} won't be sent to SAP because it's not from {string.Join(", ", _sapConfig.SapServiceConfigs.Select(x => x.CountryCode))}");
                return "Invalid billing country value.";
            }
            if (String.IsNullOrEmpty(dopplerUser.FederalTaxID))
            {
                _logger.LogInformation($"{dopplerUser.Email} won't be sent to SAP because it doesn't have a cuit value");
                return "Invalid cuit value.";
            }
            if (!dopplerUser.PlanType.HasValue)
            {
                _logger.LogInformation($"{dopplerUser.Email} won't be sent to SAP because it doesn't have a plan type id");
                return "Invalid plan type value.";
            }

            return string.Empty;
        }
```
y en caso de un error retorna una ```ValidationException```:
```
var userVerificationError = VerifyUserInformation(dopplerUser);
if (!string.IsNullOrEmpty(userVerificationError))
{
      throw new ValidationException(userVerificationError);
}
```

Y del lado del controllador solo llamamos al metodo y manejamos de manera especial cuando venga un ValidationException:
```
catch (ValidationException e)
 {
      _logger.LogError(e, $"Failed at creating/updating user: {dopplerUser.Id}. Because the user has a validation error: 
      {e.Message}");
       return new BadRequestObjectResult(e.Message);
}
```

Que piensan?.

Reply:
Done